File: app/src/main/java/com/shindra/NosTrams/StartActivity.java
package com.shindra.NosTrams;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.MyViewModel;
import com.shindra.NosTrams.NosTramsAdapter;
import com.shindra.R;
import com.shindra.Timetable.TimetableActivity;
import com.shindra.arrakis.observable.ObservableExtensionKt;
import com.shindra.arrakis.observable.ObservableListener;
import com.shindra.ctslibrary.apibo.RouteType;
import com.shindra.ctslibrary.bo.Line;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class StartActivity extends AppCompatActivity {

    private ArrayList<Line> nosTramsList = new ArrayList<>();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onStart() {
        super.onStart();

        setContentView(R.layout.activity_main);
        getSupportActionBar().setTitle(getString(R.string.LabelNosTrams));
        NosTramsAdapter.RecyclerTimetableClick callBack = new NosTramsAdapter.RecyclerTimetableClick() {
            @Override
            public void onTimetableClick(Line image) {
                Log.i("Info","Bouton horaire actionné");
                //setContentView(R.layout.horaire_item);
                openTimetableActivity();
                //startActivity(new Intent(StartActivity.this, TimetableActivity.class));
            }
        };

        RecyclerView nosTramList = findViewById(R.id.recyclerView);
        nosTramList.setLayoutManager (new LinearLayoutManager (this));
        RecyclerView.Adapter LinesAdapter = new NosTramsAdapter(nosTramsList, callBack);

        nosTramList.setAdapter (LinesAdapter);

        MyViewModel model = new ViewModelProvider(this).get(MyViewModel.class);
        ObservableExtensionKt.observe(model.lines(), new ObservableListener<ArrayList<Line>>() {

            @Override
            public void onLoading() {
                //call once we started the network called. Indicate that the network call is in progress
            }

            @Override
            public void onSuccess(ArrayList<Line> data) {
                //call once the network call has responded with a success
                for (Line LignesCTS : data)
                {
                    if (LignesCTS.getRouteType() == RouteType.TRAM)
                    {
                        nosTramsList.add(LignesCTS);
                    }
                    Log.i("NosTram","Recupération des Lignes terminées");
                }

                nosTramList.setAdapter(new NosTramsAdapter(nosTramsList, callBack));
            }
            @Override
            public void onError(@NotNull Throwable throwable) {
                //call if the network call has responded with an error
            }
        });

    }
    public void openTimetableActivity(){
        Intent intent = new Intent (this, TimetableActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/shindra/tram/StartActivity.kt
package com.shindra.tram



//import com.shindra.utilis.MyViewModel.lines
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.shindra.utilis.LoadingClass
import com.shindra.utilis.MyViewModel
import com.shindra.R
import com.shindra.time.TimeActivity
import com.shindra.arrakis.extension.toArrayList
import com.shindra.arrakis.observable.ObservableListener
import com.shindra.arrakis.observable.observe
import com.shindra.ctslibrary.apibo.RouteType
import com.shindra.ctslibrary.bo.Line
import kotlin.collections.ArrayList


class StartActivity : AppCompatActivity(), TramAdapter.TramViewHolder.RecyclerLineClick {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        title = getString(R.string.tram)
        val listInit = ArrayList<Line>()

        val lineRecyclerView = findViewById<RecyclerView>(R.id.recycler_view)
        lineRecyclerView.adapter = TramAdapter(listInit, this)
        lineRecyclerView.layoutManager = LinearLayoutManager(this)
        lineRecyclerView.setHasFixedSize(true)
        val dialog = LoadingClass(this)
        val model = ViewModelProvider(this).get(MyViewModel::class.java)
        model.lines().observe(object : ObservableListener<ArrayList<Line>> {
            override fun onLoading() {
                //call once we started the network called. Indicate that the network call is in progress
                dialog.show()
            }

            override fun onSuccess(data: ArrayList<Line>) {
                dialog.dismiss()
                //call once the network call has responded with a success
                (lineRecyclerView.adapter as TramAdapter).lineList = data.filter{it.routeType == RouteType.TRAM }.toArrayList()!!
                (lineRecyclerView.adapter as TramAdapter).notifyDataSetChanged()
            }

            override fun onError(throwable: Throwable) {
                //call if the network call has responded with an error
            }
        })
    }

    override fun onLineClick(line: Line) {
        //val intent = TimeActivity.newInstance(this,line.name)
        val intent = Intent(this, TimeActivity ::class.java)
        intent.putExtra("name", line.name)
        startActivity(intent)
    }

}